### PR TITLE
Adding a log pipe buffer size configuration test

### PR DESCRIPTION
**Pull request by bneradt**

### CODE REVIEW

File: tests/gold_tests/logging/log_pipe.test.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+import os
+
+Test.Summary = '''
+Test custom log file format
+'''
+# This test requires Curl and is Linux specific.
+Test.SkipUnless(
+    Condition.HasProgram(
+        "curl", "Curl need to be installed on system for this test to work"),
```
Comment (shinrich) [inline tests/gold_tests/logging/log_pipe.test.py:27]: As I recall they were removing the curl checks.  So we can assume any testing system has curl.

Comment (randall) [reply tests/gold_tests/logging/log_pipe.test.py:27]: This was part of 6c1a90f4f209a02cf1345ac4c20e2e67c4c58db1:
`The checks for grep, sed, curl and nc are superflous. Most systems will have them,
    and if not, we should just fail hard, and it's obvious what the user has to do.
`

**Comment by bneradt** [comment]
[approve ci autest]

File: tests/gold_tests/logging/log_pipe.test.py (hunk 1 above)
Comment (bneradt) [reply tests/gold_tests/logging/log_pipe.test.py]: Thanks for pointing this out, @shinrich and @randall . I updated the PR accordingly.

**Comment by bneradt** [comment]
[approve ci autest]

**Comment by SolidWallOfCode** [comment]
Seems reasonable.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**